packages/vite-plugin-cloudflare/src/miniflare-options.ts

```diff
@@ -197,12 +214,73 @@ export function getDevMiniflareOptions(
 	viteDevServer: vite.ViteDevServer
 ): MiniflareOptions {
 	const resolvedViteConfig = viteDevServer.config;
+	const logger = new ViteMiniflareLogger(resolvedViteConfig);
+
 	const entryWorkerConfig = getEntryWorkerConfig(resolvedPluginConfig);
 	const assetsConfig =
 		resolvedPluginConfig.type === "assets-only"
 			? resolvedPluginConfig.config.assets
 			: entryWorkerConfig?.assets;
 
+	const headersFile = path.join(resolvedViteConfig.publicDir, HEADERS_FILENAME);
+	const redirectsFile = path.join(
+		resolvedViteConfig.publicDir,
+		REDIRECTS_FILENAME
+	);
+
+	watch([headersFile, redirectsFile], {
+		persistent: true,
+		ignoreInitial: true,
+	}).on("all", async () => {
+		viteDevServer.restart();
+	});
+
```

Comment:
This file watching should be moved to the `handleHotUpdate` hook here:

https://github.com/cloudflare/workers-sdk/blob/869ec7b916487ec43b958a27bdfea13588c5685f/packages/vite-plugin-cloudflare/src/index.ts#L244-L248

I think you'll need to refactor things to move most of the new logic out of this file and then pass the `parsedHeaders` and `parsedRedirects` to the `getDevMiniflareOptions` function.